gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageImplTest.java

```diff
@@ -567,4 +576,43 @@ private static List<String> getExpectedRequestsForCreateObject(
                 .collect(toList()))
         .build();
   }
+
+  @Test
+  public void initializeRequest_withGcsAuditLogEnabled() throws IOException {
+    AssertingLogHandler jsonLogHander = new AssertingLogHandler();
+    Logger jsonTracingLogger =
+        jsonLogHander.getLoggerForClass(EventLoggingHttpRequestInitializer.class.getName());
+
+    try {
+
+      String ugiUser = UUID.randomUUID().toString();
+      UserGroupInformation ugi = UserGroupInformation.createRemoteUser(ugiUser);
+      UserGroupInformation.setLoginUser(ugi);
+      GoogleCloudStorage storage =
+          new GoogleCloudStorageImpl(
+              GoogleCloudStorageTestHelper.getStandardOptionBuilder()
+                  .setGcsAuditLogEnabled(true)
+                  .setTraceLogEnabled(true)
+                  .build(),
+              GoogleCloudStorageTestHelper.getCredentials());
+
+      int expectedSize = 5 * 1024 * 1024;
+      StorageResourceId resourceId = new StorageResourceId(TEST_BUCKET, name.getMethodName());
+      writeObject(
+          storage, resourceId, /* partitionSize= */ expectedSize, /* partitionsCount= */ 1);
+
+      // 3 types of log records are present in the job handler
+      // 1. Geting the object - GET call
+      // 2. Starting object upload (First write call) - POST call
+      // 3. Writing the chunk (Handled by the library internally therefore does not carry the storage options we set for HTTP request, thererfore we get the audit logs at the initial step and not one for each chunk written) - PUT call
+      jsonLogHander.getAllLogRecords().stream().filter(logRecord -> !logRecord.get("request_method").equals("PUT")).collect(
+          Collectors.toList()).forEach(logRecord -> {
+            LinkedTreeMap<String, Object> requestHeaders = (LinkedTreeMap<String, Object>) logRecord.get("request_headers");
+            assertEquals(ugiUser, requestHeaders.get("x-goog-custom-audit-user"));
```

Comment:
nit: Can we use this from gcsIml. CUSTOM_AUDIT_USER_HEADER ?

Reply:
Done.